### DISCUSSION TITLE
Why Lucid method create doesn't TS check for payload completeness

### DISCUSSION THREAD

**Discussion by pinkynrg** [adonisjs/core#3978] [Help] [answered]
I'm creating a migration script from old db to an adonis connected db. 

When I do my SomeModel.create({...}) i don't get any warnings if the given payload is incomplete. 

I was about to override the definition of BaseModel create method from this:

`create<T extends LucidModel>(this: T, values: Partial<ModelAttributes<InstanceType<T>>>, options?: ModelAssignOptions): Promise<InstanceType<T>>;`

to something like this:

`create<T extends LucidModel>(this: T, values: ModelAttributes<InstanceType<T>>, options?: ModelAssignOptions): Promise<InstanceType<T>>;`

why isn't this a default behavior for the create api?

**Comment by thetutlage** [comment, marked as answer]
If my incomplete you mean the values are missing, then it is perfectly okay to not define all the attributes in a save call.

- Many times you will have defaults defined within the database.
- Some times you will use hooks to generate some values.
